tests/unit/test_synapse_api.py

```diff
@@ -102,12 +102,45 @@ def test_get_nonce_success(mock_requests):
 
 def test_get_nonce_error(monkeypatch: pytest.MonkeyPatch):
     """
-    arrange: set User parameters.
-    act: register the user.
+    arrange: none.
+    act: get nonce.
     assert: NetworkError is raised.
     """
     mock_response_error = requests.exceptions.ConnectionError("Connection error")
     mock_response = mock.Mock(side_effect=mock_response_error)
     monkeypatch.setattr("synapse.api.requests.get", mock_response)
     with pytest.raises(synapse.APIError, match="Failed to request"):
         synapse.api._get_nonce()
+
+
+@mock.patch("synapse.api.Session")
+def test_get_version_success(mock_session):
```

Comment:
Perhaps IO missed, it, there seems to be 2 cases, one where the regex matches and one where it doesn't, should there be a test for both cases?

Reply:
Good point, I added a new unit test for that.